src/service/layer/layer_fn.rs

```diff
@@ -75,7 +74,11 @@ mod tests {
                 ctx: Context<State>,
                 req: Request,
             ) -> impl Future<Output = Result<Self::Response, Self::Error>> + Send + '_ {
-                self.0.serve(ctx, req).map_ok(|res| res.to_uppercase())
+                let serve = self.0.serve(ctx, req);
```

Comment:
valid but you can just implement it as `async fn serve(...) -> Result<...> { ... }`, is more rama-idiomatic, and easier. Then you do not need the `async move`.